src/TestLogger.php

```diff
@@ -153,10 +174,11 @@ public function hasRecordThatPasses(callable $predicate, string $level): bool
      */
     public function __call(string $method, array $args): bool
     {
-        if (\preg_match('/(.*)(Debug|Info|Notice|Warning|Error|Critical|Alert|Emergency)(.*)/', $method, $matches) > 0) {
+        $levelNames = \implode('|', \array_map('ucfirst', \array_keys($this->levelMap)));
+        if (\preg_match('/(.*)(' . $levelNames . ')(.*)/', $method, $matches) > 0) {
```

Comment:
I'm guessing the idea behind this change is that a user may have instantiated the `TestLogger` without defining all of those levels in their map?  If so, I think my preference would be to revert this change and add some validation to the constructor to ensure that the map contains all log levels.  That would ensure that all of the magic methods defined on lines 15-54 are always available and guaranteed to never throw a `BadMethodCallException`.

Reply:
Actually this was motivated by DRY. It smelled to me to repeat the name information twice in the class. I have kept this code, but added the requested validation to the __construct() method. Happy to remove this code if you prefer the simplicity of explicit repetition instead of DRY.